runtime/src/software_interrupt.c

```diff
@@ -47,13 +47,14 @@ software_interrupt_deferred_sigalrm_max_print()
 		printf("Worker %d: %d\n", i, software_interrupt_deferred_sigalrm_max[i]);
 	}
 	fflush(stdout);
+	if (software_interrupt_deferred_sigalrm_max) free((void *)software_interrupt_deferred_sigalrm_max);
```

Comment:
Kinda weird to use print to free. Maybe create `software_interrupt_cleanup` and call that after `software_interrupt_deferred_sigalrm_max_print` in `runtime_cleanup`?

Reply:
Yeah, I hesitated whether I should create a separate function for just one call. But agreed, this is not good SE. So, I'll do what you said.

Reply:
The cast to `(void *)` is quite strange to me. Is it required to remove a warning? If it is superfluous, then I'd remove it. I don't think that it helps document the use, and it is far enough away from my sense of typical C that I've seen that it feels like an outlier.

Reply:
Yes, I did it to remove the warning. I think it is because of its type that is unusual (volatile _Atomic(sig_atomic_t) ). Any other better way to do it without casting?